### ISSUE TITLE
Help to cache database in Elixir app

### ISSUE THREAD

**Issue by petrus-jvrensburg** [g-andrade/locus#47]
I'm struggling to get the cache behavior to work in an Elixir app: the app seems to pull the database from the network each time if I set it up like in the README, without any cache file specified. 

But if I try to specify the cache file explicitly, I get an error:

```
iex> :locus.start_loader(:city, {:maxmind, "GeoLite2-City"}, [{:database_cache_file, "/tmp/maxmind_city.mmdb.gz"}])
{:error, {:invalid_opt, {:database_cache_file, "/tmp/maxmind_city.mmdb.gz"}}}
```

Any idea what I'm doing wrong here?

**Comment by g-andrade** [comment]
Hi Petrus, your example worked fine for me. Does the folder "/tmp/" exist on the environment you tried? It's the only source of `invalid_opt` I can think of (and admittedly not very graceful):
https://github.com/g-andrade/locus/blob/329f5c918abe63f70ede2edc8fa52946e6d89e84/src/locus_loader.erl#L342-L349

**Comment by petrus-jvrensburg** [comment]
Could it be the format of the input arguments? Otherwise, I suspect the issue might be in the extension checking. What I'm seeing is:

```
iex(6)> :locus.start_loader(:city, {:maxmind, "GeoLite2-City"}, [{:database_cache_file, "/tmp/maxmind_city.mmdb.gz"}])
{:error, {:invalid_opt, {:database_cache_file, "/tmp/maxmind_city.mmdb.gz"}}}
iex(7)> :filename.dirname("/tmp/maxmind_city.mmdb.gz")
"/tmp"
iex(8)> :filelib.is_dir("/tmp")
true
```

I tried a few different formats for the arguments, but without luck:

A tuple:
```
iex(8)> :locus.start_loader(:city, {:maxmind, "GeoLite2-City"}, [{:database_cache_file, "/tmp/maxmind_city.mmdb.gz"}])
{:error, {:invalid_opt, {:database_cache_file, "/tmp/maxmind_city.mmdb.gz"}}}
```
A map with an atom key:
```
iex(9)> :locus.start_loader(:city, {:maxmind, "GeoLite2-City"}, [%{database_cache_file: "/tmp/maxmind_city.mmdb.gz"}])
{:error, {:invalid_opt, %{database_cache_file: "/tmp/maxmind_city.mmdb.gz"}}}
```
A map with a string key:
```
iex(10)> :locus.start_loader(:city, {:maxmind, "GeoLite2-City"}, [%{"database_cache_file" => "/tmp/maxmind_city.mmdb.gz"}])
```

**Comment by g-andrade** [comment]
Ahh, it's the path! It's expecting an Erlang string (charlist). I'll fix it. This should work in the mean time:

```elixir
:locus.start_loader(:city, {:maxmind, "GeoLite2-City"}, database_cache_file: ~c"/tmp/maxmind_city.mmdb.gz"])
```

**Comment by g-andrade** [comment]
> Ahh, it's the path!

Fixed in 49ef423, released as 2.3.10. The following should now also work:

```elixir
:locus.start_loader(:city, {:maxmind, "GeoLite2-City"}, database_cache_file: "/tmp/maxmind_city.mmdb.gz"])
```

Let me know if it's not so. Thanks for reporting the issue!

**Comment by petrus-jvrensburg** [comment]
Thanks! 

The workaround worked:

```
iex(1)> :locus.start_loader(:city, {:maxmind, "GeoLite2-City"}, [database_cache_file: ~c"/tmp/maxmind_city.mmdb.gz"])
:ok
```

Also, I checked that in 2.3.10 it works with regular strings:
```
iex(1)> :locus.start_loader(:city, {:maxmind, "GeoLite2-City"}, database_cache_file: "/tmp/maxmind_city.mmdb.gz")
:ok
```
